test/widgets/poll_test.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/api/model/submessage.dart';
+import 'package:zulip/model/narrow.dart';
+import 'package:zulip/model/store.dart';
+import 'package:zulip/widgets/message_list.dart';
+import 'package:zulip/widgets/poll.dart';
+
+import '../api/fake_api.dart';
+import '../example_data.dart' as eg;
+import '../model/binding.dart';
+import '../model/message_list_test.dart';
+import '../model/test_store.dart';
+import 'test_app.dart';
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  late PerAccountStore store;
+
+  Future<void> prepare({Iterable<User>? users}) async {
+    addTearDown(testBinding.reset);
+
+    await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+    store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+    await store.addUsers(users ?? [eg.selfUser, eg.otherUser]);
+  }
+
+  Future<void> prepareZulipWidget(WidgetTester tester,
+    Object? submessageContent,
+    {Iterable<(User, int)> voterOptionIndexPairs = const []}
+  ) async {
+    final message = eg.streamMessage(id: 123, sender: eg.selfUser,
+      submessages: [
+        eg.submessage(messageId: 123, content: submessageContent),
+      ]);
+    (store.connection as FakeApiConnection).prepare(json:
+      newestResult(foundOldest: true, messages: [message]).toJson());
+    await tester.pumpWidget(TestZulipApp(accountId: eg.selfAccount.id,
+      child: MessageListPage(narrow: TopicNarrow.ofMessage(message))));
+
+    await tester.pumpAndSettle();
+
+    for (final (voter, optionIndex) in voterOptionIndexPairs) {
+      await store.handleEvent(eg.submessageEvent(123, voter.userId,
+        content: PollVoteEvent(
+          key: PollEvent.optionKey(senderId: null, optionIndex: optionIndex),
+          op: VoteOp.add).toJson()));
+    }
+    await tester.pump();
+  }
+
+  Finder findInPoll(Finder matching) => find.descendant(
+    of: find.byType(PollWidget),
+    matching: matching,
+  );
+
+  testWidgets('smoke', (tester) async {
+    await prepare();
+    await prepareZulipWidget(tester,
+      eg.pollWidgetDataJson,
+      voterOptionIndexPairs: [
+        (eg.selfUser, 0),
+        (eg.selfUser, 1),
+        (eg.otherUser, 1),
+      ]
+    );
+
+    check(findInPoll(find.text('favorite letter')).evaluate()).single;
```

Comment:
It would be nice if the test-setup code made it clearer why this is the string we want to look for. When I searched for this string in the file, I didn't get any results besides this one.

Could maybe:

- make `eg.pollWidgetDataJson` a function, and callers would pass the question they want. (So, searching 'favorite letter' in this file would identify that callsite.)
- instead of using the string literal 'favorite letter', use the value by reference, like `eg.pollWidgetDataJson['extra_data']['question']`
- rename `eg.pollWidgetDataJson` to something like `eg.pollWidgetDataJsonFavoriteLetter`, so the connection with the test-setup code is clearer, and also so the definition in example_data.dart gives a clue that its `question` property is a particular value that some tests will rely on